src/DotVVM.Framework/Controls/DotvvmControl.cs

```diff
@@ -435,6 +457,19 @@ public static bool IsNamingContainer(DotvvmBindableObject control)
             return (bool)control.GetValue(Internal.IsNamingContainerProperty)!;
         }
 
+        /// <summary>
+        /// Determines whether the specified control is a naming container.
+        /// </summary>
+        protected IRenderAdapter? GetRenderAdapter(IDotvvmRequestContext context)
+        {
+            var routeRouteName = context.Route.RouteName;
+
+            if (!context.Configuration.ExperimentalFeatures.ControlRenderAdapters.IsEnabledForRoute(routeRouteName))
+                return null;
+
+            return (IRenderAdapter?)GetValueRaw(Internal.RenderAdapterProperty);
```

Comment:
If we want to merge, I'll have to resuscitate the benchmarks :/ This is one of the hottest paths in DotVVM and checking the `IsEnabledForRoute` for every control does not seem like a terribly good idea.

Reply:
In general, I don't think we need to hide behind an experimental feature. It's not going to do anything, unless something starts using it.

Reply:
Actually the main point for using ExperimentalFeatures flag was to limit this feature to only some routes because I was worried about performance of getting value of Internal.RenderAdapterProperty